dessin-pdf/src/lib.rs

```diff
@@ -223,43 +235,60 @@ impl Exporter for PDFExporter {
         Ok(())
     }
 
-    fn export_text(&mut self, text: TextPosition) -> Result<(), Self::Error> {
-        let font = text
-            .font
-            .as_ref()
-            .map(|f| f.font_family())
-            .unwrap_or("default");
-        let font = self
-            .fonts
-            .get(font)
-            .and_then(|font| match text.font_weight {
-                FontWeight::Regular => Some(font.regular.clone()),
-                FontWeight::Bold => font.bold.clone(),
-                FontWeight::Italic => font.italic.clone(),
-                FontWeight::BoldItalic => font.bold_italic.clone(),
-            })
-            .unwrap();
+    fn export_text(
+        &mut self,
+        TextPosition {
+            text,
+            align: _,
+            font_weight,
+            on_curve: _,
+            font_size,
+            reference_start,
+            direction,
+            font,
+        }: TextPosition,
+    ) -> Result<(), Self::Error> {
+        let font = font.clone().unwrap_or(FontRef::default());
+
+        // search if (font_ref, font_weight) is stocked in used_font
+        if !self.used_font.contains_key(&(font.clone(), font_weight)) {
+            self.used_font.insert(
+                (font.clone(), font_weight),
+                match font::get(font.clone()).get(font_weight) {
+                    dessin::font::Font::OTF(b) | dessin::font::Font::TTF(b) => {
+                        self.doc.add_external_font(b.as_slice())?
+                    }
+                },
+            );
+        }
+
+        let will_survive = font::get(font);
+        let font = will_survive.get(font_weight);
+        let font: IndirectFontRef = match font {
+            dessin::font::Font::OTF(b) | dessin::font::Font::TTF(b) => {
+                self.doc.add_external_font(b.as_slice())?
+            }
+        };
```

Comment:
Tu ajoutes dans tous les cas la font.

```suggestion
        let font = self.used_font.entry((font.clone(), font_weight)).or_insert_with(|| 
                 match font::get(font.clone()).get(font_weight) {
                    dessin::font::Font::OTF(b) | dessin::font::Font::TTF(b) => {
                        self.doc.add_external_font(b.as_slice())?
                    }
                },
        )
```

Reply:
Normalement, c'est régler.
Si c'est bon, j'enlèverai les commentaires et ça devrait être bon